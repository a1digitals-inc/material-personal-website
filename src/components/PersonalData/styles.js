import { makeStyles } from "@material-ui/core/styles";

const useStyles = makeStyles((theme) => ({
  root: {
    display: "flex",
    flexDirection: "column",
    alignItems: "center",
    textAlign: "center",
  },
  avatar: {
    margin: theme.spacing(2),
    width: theme.spacing(18),
    height: theme.spacing(18),
  },
  text: {
    position: "relative",
    maxWidth: 720,
    textAlign: "center",
    "& > *": {
      marginBottom: theme.spacing(2),
    },
  },
}));

export default useStyles;
