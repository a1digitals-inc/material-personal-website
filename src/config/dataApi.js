import defaultprofilepic from "../assets/images/defaultprofilepic.jfif";
import * as portfolioTypes from "../components/Cards/constants";

export const personalData = {
  greeting: "Hi, I'm",
  name: "Dyarlen Iber",
  description:
    "I'm a Full-Stack developer with professional experience working with PHP, Laravel, JavaScript, Node.js, SQL, React and Go. Take a look at my work or get in touch!",
  email: "dyarlen1@gmail.com",
  profilePicture: defaultprofilepic,
  socialMedias: [
    {
      name: "github",
      url: "https://github.com/dyarleniber",
    },
    {
      name: "linkedin",
      url: "https://ie.linkedin.com/in/dyarleniber",
    },
    {
      name: "instagram",
      url: "https://www.instagram.com/dyarleniber/",
    },
    {
      name: "dev",
      url: "https://dev.to/dyarleniber",
    },
  ],
};

export const menuItems = [
  {
    to: "/",
    primary: "Home",
  },
  {
    to: "/portfolio",
    primary: "Portfolio",
  },
];

export const portfolioData = {
  title: "Portfolio",
  description:
    "All my projects are covered by automated tests and deployed on platforms such as GitHub Pages or Netlify. For more details about the source code, automated tests, CI/CD workflow and technologies used, please visit the github page of each project.",
};

const frontEndPortfolio = [
  {
    id: 101,
    name: "Personal website",
    description:
      "ðŸ’» Personal website using Javascript, React, Hooks, Material-UI and Jest.",
    type: portfolioTypes.FRONTEND_TYPE,
    technologies: ["Javascript", "React", "Hooks", "Material-UI", "Jest"],
    homepage: "https://www.dyarleniber.com",
    github: "https://github.com/dyarleniber/personal-website",
  },
  {
    id: 102,
    name: "New York Times movie reviews",
    description:
      "Application for search New York Times movie reviews and get movie critics using Javascript, React, Redux, React hooks, redux-saga and styled-components.",
    type: portfolioTypes.FRONTEND_TYPE,
    technologies: [
      "React",
      "Redux",
      "Hooks",
      "redux-saga",
      "styled-components",
    ],
    homepage: null,
    github: "https://github.com/dyarleniber/nyt-movie-reviews",
  },
  {
    id: 103,
    name: "Habit tracker front-end",
    description:
      "Habit tracker front-end system using Javascript, React, Redux, React hooks, redux-saga and styled-components.",
    type: portfolioTypes.FRONTEND_TYPE,
    technologies: [
      "React",
      "Redux",
      "Hooks",
      "redux-saga",
      "styled-components",
    ],
    homepage: null,
    github: "https://github.com/dyarleniber/habit-tracker-front-end",
  },
  {
    id: 104,
    name: "Smart brain",
    description: "Udemy Final Project.",
    type: portfolioTypes.FRONTEND_TYPE,
    technologies: ["React", "Redux"],
    homepage: null,
    github: "https://github.com/dyarleniber/smart-brain",
  },
  {
    id: 105,
    name: "To-do list",
    description: "Simple To-do list with React, Redux and Hooks.",
    type: portfolioTypes.FRONTEND_TYPE,
    technologies: ["React", "Redux", "Hooks", "Bootstrap"],
    homepage: null,
    github: "https://github.com/dyarleniber/react-to-do-list",
  },
];

const backEndPortfolio = [
  {
    id: 201,
    name: "Habit tracker back-end",
    description:
      "Habit tracker back-end system using Javascript, Node.js, Express, JWT, MongoDB, Mongoose, Nodemailer, Handlebars, Redis, Winston and Jest with 100% code coverage in tests.",
    type: portfolioTypes.BACKEND_TYPE,
    technologies: ["Node.js", "Express", "JWT", "MongoDB", "Redis", "Jest"],
    homepage: null,
    github: "https://github.com/dyarleniber/habit-tracker-back-end",
  },
  {
    id: 202,
    name: "Payment API",
    description: "Payment gateway API.",
    type: portfolioTypes.BACKEND_TYPE,
    technologies: ["PHP"],
    homepage: null,
    github: "https://github.com/dyarleniber/php-payment-api",
  },
  {
    id: 203,
    name: "Smart brain API",
    description: "Final project for Udemy course.",
    type: portfolioTypes.BACKEND_TYPE,
    technologies: ["Node.js", "Express", "JWT", "Redis"],
    homepage: null,
    github: "https://github.com/dyarleniber/smart-brain-api",
  },
  {
    id: 204,
    name: "PHP Docker environment",
    description: "PHP and MySQL environment with Docker and Docker Compose.",
    type: portfolioTypes.BACKEND_TYPE,
    technologies: ["PHP", "MySQL", "Docker"],
    homepage: null,
    github: "https://github.com/dyarleniber/docker-php",
  },
  {
    id: 205,
    name: "Awesome Fizzbuzz",
    description:
      "This project was developed applying the Chain of Responsibility Pattern and the Builder Pattern to develop FizzBuzz.",
    type: portfolioTypes.BACKEND_TYPE,
    technologies: ["PHP", "Docker"],
    homepage: null,
    github: "https://github.com/dyarleniber/awesome-fizzbuzz",
  },
];

export const portfolio = [...backEndPortfolio, ...frontEndPortfolio];
