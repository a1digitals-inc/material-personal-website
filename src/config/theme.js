import { createMuiTheme, responsiveFontSizes } from "@material-ui/core/styles";
import grey from "@material-ui/core/colors/grey";

const blackColor = "#000";
const whiteColor = "#fff";

let theme = createMuiTheme({
  palette: {
    primary: {
      light: grey[300],
      main: grey[500],
      dark: grey[700],
    },

    secondary: {
      light: grey[300],
      main: grey[500],
      dark: grey[700],
    },

    text: {
      primary: blackColor,
      secondary: blackColor,
      disabled: blackColor,
    },

    action: {
      active: blackColor,
    },

    background: {
      default: grey[100],
      paper: whiteColor,
    },
  },
});

theme = responsiveFontSizes(theme);

export default theme;
