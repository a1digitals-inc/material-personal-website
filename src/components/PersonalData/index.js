import React from "react";
import Grid from "@material-ui/core/Grid";
import Avatar from "@material-ui/core/Avatar";
import Typography from "@material-ui/core/Typography";
import IconButton from "@material-ui/core/IconButton";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faEnvelope } from "@fortawesome/free-solid-svg-icons";

import { personalData } from "../../config/dataApi";
import useStyles from "./styles";
import ButtonLink from "../ButtonLink";

const PersonalData = () => {
  const classes = useStyles();

  const {
    name,
    description,
    email,
    profilePicture,
    socialMedias,
  } = personalData;

  return (
    <div className={classes.paper}>
      <Avatar
        alt="profile picture"
        src={profilePicture}
        className={classes.avatar}
      />
      <Typography variant="h5" gutterBottom>
        {name}
      </Typography>
      <Typography variant="body2" gutterBottom>
        {description}
      </Typography>
      <Typography variant="overline" display="block" gutterBottom>
        {email}
      </Typography>
      <Grid container className={classes.socialmedia} spacing={1}>
        <Grid item xs={12}>
          <Grid container justify="center" spacing={1}>
            <Grid item>
              <IconButton
                aria-label="email"
                component="a"
                href={`mailto:${email}`}
              >
                <FontAwesomeIcon icon={faEnvelope} />
              </IconButton>
            </Grid>
            {socialMedias.map((socialMedia) => (
              <Grid item key={socialMedia.name}>
                <IconButton
                  aria-label={socialMedia.name}
                  target="_blank"
                  component="a"
                  href={socialMedia.url}
                >
                  <FontAwesomeIcon icon={["fab", socialMedia.name]} />
                </IconButton>
              </Grid>
            ))}
          </Grid>
        </Grid>
        <Grid item xs={12}>
          <ButtonLink
            variant="outlined"
            to={"/portfolio"}
            primary={"See my portfolio"}
          />
        </Grid>
      </Grid>
    </div>
  );
};

export default PersonalData;
