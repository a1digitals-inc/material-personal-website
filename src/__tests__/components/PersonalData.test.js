import React from "react";
import { shallow } from "enzyme";

import * as data from "../../config/dataApi";
import "../../config/icons";
import PersonalData from "../../components/PersonalData";

const originalPersonalData = data.personalData;

beforeAll(() => {
  delete data.personalData;
  data.personalData = {
    greeting: "greeting",
    name: "name",
    description: "description",
    email: "email",
    profilePicture: "profilePicture",
    socialMedias: [
      {
        name: "socialMediaName",
        url: "socialMediaUrl",
      },
    ],
  };
});

afterAll(() => {
  data.personalData = originalPersonalData;
});

describe("PersonalData component", () => {
  it("should render without crashing", () => {
    const wrapper = shallow(<PersonalData />);
    expect(wrapper).toMatchSnapshot();
  });
});
