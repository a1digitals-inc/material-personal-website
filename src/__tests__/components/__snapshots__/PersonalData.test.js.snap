// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PersonalData component should render without crashing 1`] = `
<div
  className="makeStyles-root-1"
>
  <WithStyles(ForwardRef(Avatar))
    alt="Profile picture"
    className="makeStyles-avatar-2"
    src="profilePicture"
  />
  <WithStyles(ForwardRef(Grid))
    className="makeStyles-text-3"
    container={true}
  >
    <WithStyles(ForwardRef(Grid))
      item={true}
      xs={12}
    >
      <WithStyles(ForwardRef(Typography))
        variant="h2"
      >
        greeting name
      </WithStyles(ForwardRef(Typography))>
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      item={true}
      xs={12}
    >
      <WithStyles(ForwardRef(Typography))
        variant="h5"
      >
        description
      </WithStyles(ForwardRef(Typography))>
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      item={true}
      xs={12}
    >
      <WithStyles(ForwardRef(Typography))
        variant="button"
      >
        email
      </WithStyles(ForwardRef(Typography))>
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      item={true}
      xs={12}
    >
      <WithStyles(ForwardRef(IconButton))
        aria-label="email"
        component="a"
        href="mailto:email"
      >
        <MailIcon />
      </WithStyles(ForwardRef(IconButton))>
      <WithStyles(ForwardRef(IconButton))
        aria-label="socialMediaName"
        component="a"
        href="socialMediaUrl"
        key="socialMediaName"
        target="_blank"
      >
        <FontAwesomeIcon
          border={false}
          className=""
          fixedWidth={false}
          flip={null}
          icon={
            Array [
              "fab",
              "socialMediaName",
            ]
          }
          inverse={false}
          listItem={false}
          mask={null}
          pull={null}
          pulse={false}
          rotation={null}
          size={null}
          spin={false}
          swapOpacity={false}
          symbol={false}
          title=""
          transform={null}
        />
      </WithStyles(ForwardRef(IconButton))>
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      item={true}
      xs={12}
    >
      <ButtonLink
        primary="See my portfolio"
        to="/portfolio"
      />
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
</div>
`;
